client/components/EventsPlanningFilters/EditFilter.tsx

```diff
@@ -51,6 +51,25 @@ export class EditFilter extends React.Component<IEventsPlanningContentPanelProps
         this.popupContainer = React.createRef();
     }
 
+    componentDidUpdate(prevProps: IEventsPlanningContentPanelProps) {
+        if (prevProps.filter?._id !== this.props.filter?._id) {
+            const filter = this.props.filter != null ?
+                cloneDeep(this.props.filter) :
+                eventPlanningUtils.defaultFilterValues();
+
+            // eslint-disable-next-line react/no-did-update-set-state
+            this.setState({
+                pristine: true,
+                filter: filter,
+                invalid: false,
+                errors: {},
+                profile: this.getProfile(filter.item_type),
+            }, () => {
+                this.props.onPristineChange?.(true);
+            });
+        }
+    }
+
```

Comment:
I'm wondering if instead of this change we could use a `key` prop to reset the state without the lifecycle tricks. You know what I mean? Thoughts?